Cloudflare provider having issues

I've redacted my personal details.

I get the following error when running `terraformer import cloudflare --resources=dns`

```
2019/07/29 13:14:14 Refreshing state... cloudflare_record.MX_mydomain--com--au_<hash_id>
panic: interface conversion: interface {} is nil, not string

goroutine 1 [running]:
github.com/GoogleCloudPlatform/terraformer/providers/cloudflare.(*DNSGenerator).PostConvertHook(0xc000168190, 0x3f44ec0, 0xc000168190)
	/Users/apetrovic/code/terraformer/providers/cloudflare/dns.go:118 +0x485
github.com/GoogleCloudPlatform/terraformer/cmd.Import(0x3f3e9c0, 0xc0000cde20, 0xc0004ae970, 0x1, 0x1, 0x3a079a8, 0x1e, 0x39d4638, 0x9, 0x39cdf5f, ...)
	/Users/apetrovic/code/terraformer/cmd/import.go:106 +0x23c
github.com/GoogleCloudPlatform/terraformer/cmd.newCmdCloudflareImporter.func1(0xc0003be280, 0xc0004ae950, 0x0, 0x1, 0x0, 0x0)
	/Users/apetrovic/code/terraformer/cmd/cloudflare.go:30 +0xe4
github.com/spf13/cobra.(*Command).execute(0xc0003be280, 0xc0004ae930, 0x1, 0x1, 0xc0003be280, 0xc0004ae930)
	/Users/apetrovic/code/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:762 +0x465
github.com/spf13/cobra.(*Command).ExecuteC(0xc00017d900, 0x39cfda3, 0x6, 0xc000541f88)
	/Users/apetrovic/code/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:852 +0x2ec
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/apetrovic/code/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:800
github.com/GoogleCloudPlatform/terraformer/cmd.Execute(0x0, 0x0)
	/Users/apetrovic/code/terraformer/cmd/root.go:36 +0x28
main.main()
	/Users/apetrovic/code/terraformer/main.go:25 +0x22
```

--- a/providers/cloudflare/dns.go
+++ b/providers/cloudflare/dns.go
@@ -44,6 +44,7 @@ func (*DNSGenerator) createZonesResource(api *cf.API, zoneID string) ([]terrafor
 		[]string{},
 		map[string]string{},
 	)
+	resource.IgnoreKeys = append(resource.IgnoreKeys, "^meta$")
 
 	return []terraform_utils.Resource{resource}, nil
 }
@@ -57,7 +58,7 @@ func (*DNSGenerator) createRecordsResources(api *cf.API, zoneID string) ([]terra
 	}
 
 	for _, record := range records {
-		resources = append(resources, terraform_utils.NewResource(
+		r := terraform_utils.NewResource(
 			record.ID,
 			fmt.Sprintf("%s_%s_%s", record.Type, record.ZoneName, record.ID),
 			"cloudflare_record",
@@ -65,10 +66,14 @@ func (*DNSGenerator) createRecordsResources(api *cf.API, zoneID string) ([]terra
 			map[string]string{
 				"zone_id": zoneID,
 				"domain":  record.ZoneName,
+				"name":    record.Name,
 			},
 			[]string{},
 			map[string]string{},
-		))
+		)
+
+		r.IgnoreKeys = append(r.IgnoreKeys, "^metadata")
+		resources = append(resources, r)
 	}
 
 	return resources, nil
@@ -102,30 +107,23 @@ func (g *DNSGenerator) InitResources() error {
 			g.Resources = append(g.Resources, tmpRes...)
 		}
 	}
-
 	g.PopulateIgnoreKeys()
 	return nil
 
 }
 
 func (g *DNSGenerator) PostConvertHook() error {
-	for i, resourceRecord := range g.Resources {
-		if resourceRecord.InstanceInfo.Type == "cloudflare_zone" {
-			continue
-		}
-
-		item := resourceRecord.Item
-		zoneID := item["zone_id"].(string)
-		for _, resourceZone := range g.Resources {
-			if resourceZone.InstanceInfo.Type != "cloudflare_zone" {
-				continue
-			}
-			if zoneID == resourceZone.InstanceState.ID {
-				if resourceRecord.InstanceInfo.Type == "cloudflare_record" {
-					g.Resources[i].Item["domain"] = "${cloudflare_zone." + resourceZone.ResourceName + ".zone}"
-				}
+	// 'record' resource have 'data' and 'value' is mutual-exclude
+	// delete which one have empty value
+	for i, resource := range g.Resources {
+		if resource.InstanceInfo.Type == "cloudflare_record" {
+			if val, ok := resource.Item["data"]; ok && len(val.(map[string]interface{})) == 0 {
+				delete(g.Resources[i].Item, "data")
+			} else if val, ok := resource.Item["value"]; ok && len(val.(string)) == 0 {
+				delete(g.Resources[i].Item, "value")
 			}
 		}
 	}
+
 	return nil
 }

